src/Api/Vault/AuthorizationHandlers/Collections/BulkCollectionAuthorizationHandler.cs

```diff
@@ -0,0 +1,205 @@
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        context.Fail();
+    }
+
+    private async Task CanReadAsync(AuthorizationHandlerContext context, CollectionOperationRequirement requirement,
+        ICollection<Collection> targetCollections, CurrentContextOrganization org)
+    {
+        if (org.Type is OrganizationUserType.Owner or OrganizationUserType.Admin ||
+            await _currentContext.ProviderUserForOrgAsync(org.Id))
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        await CheckCollectionPermissionsAsync(context, requirement, targetCollections, org, requireManagePermission: false);
+    }
+
+    private async Task CanDeleteAsync(AuthorizationHandlerContext context, CollectionOperationRequirement requirement,
+        ICollection<Collection> targetCollections, CurrentContextOrganization org)
+    {
+        // Owners, Admins, Providers, and users with DeleteAnyCollection permission can always delete collections
+        if (
+            org.Type is OrganizationUserType.Owner or OrganizationUserType.Admin ||
+            org.Permissions is { DeleteAnyCollection: true } ||
+            await _currentContext.ProviderUserForOrgAsync(org.Id))
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        // The limit collection management setting is enabled and we are not an Admin (above condition), fail
+        if (org.LimitCollectionCreationDeletion)
+        {
+            context.Fail();
+            return;
+        }
+
+        await CheckCollectionPermissionsAsync(context, requirement, targetCollections, org, requireManagePermission: true);
+    }
+
+    /// <summary>
+    /// Ensures the acting user is allowed to manage access permissions for the target collections.
+    /// </summary>
+    private async Task CanManageCollectionAccessAsync(AuthorizationHandlerContext context,
+        IAuthorizationRequirement requirement, ICollection<Collection> targetCollections, CurrentContextOrganization org)
+    {
+        // Owners, Admins, Providers, and users with EditAnyCollection permission can always manage collection access
+        if (
+            org.Permissions is { EditAnyCollection: true } ||
+            org.Type is OrganizationUserType.Owner or OrganizationUserType.Admin ||
+            await _currentContext.ProviderUserForOrgAsync(org.Id))
+        {
+            context.Succeed(requirement);
+            return;
+        }
+
+        await CheckCollectionPermissionsAsync(context, requirement, targetCollections, org, requireManagePermission: true);
+    }
+
+    private async Task CheckCollectionPermissionsAsync(
+        AuthorizationHandlerContext context,
+        IAuthorizationRequirement requirement,
+        ICollection<Collection> targetCollections,
+        CurrentContextOrganization org,
+        bool requireManagePermission)
+    {
```

Comment:
This method would be much simpler if it returned a `bool`. Then you don't have to pass in so many arguments, and the name would be easier to understand, e.g. `CanManageCollections`.